test.js

```diff
@@ -241,5 +246,37 @@ describe('shopify-token', function () {
         done();
       });
     });
+
+    it('returns an error if timeout is exceeded (no response)', function (done) {
+      scope
+        .post(pathname)
+        // Delay response header. (Client has 10ms timeout above.)
+        .delay({ head: 100 })
```

Comment:
`request.setTimeout()` only handle the case where a socket is assigned to the request and there is no activity on the socket within the specified time frame, that's why `delaySocket` works.
You also have to handle the case where a socket is not assigned to the request (`delay` in nock).
As said in https://github.com/lpinca/shopify-token/issues/4 [`timed-out`](https://github.com/floatdrop/timed-out) makes all of this very easy.